module_auto_update/models/module.py

```diff
@@ -39,36 +36,30 @@ def _compute_checksum_dir(self):
                 excluded_extensions=exclude
             )
 
-    @api.depends('state')
-    def _compute_checksum_installed(self):
-        stored_checksums = {}
-        for r in self.read(['checksum_installed']):
-            stored_checksums[r['id']] = r['checksum_installed']
-
-        for r in self:
-            if r.state == 'installed':
+    def _store_checksum_installed(self, vals):
+        if (vals.get('state') == 'installed' and
```

Comment:
I'd also check `"checksum_installed" not in vals`, to avoid double writes if the field is being manually written.